frontend/src/Pages/SignUp.js

```diff
@@ -0,0 +1,125 @@
+    }
+  }, [accessToken, history])
+
+  const onNameChange = (event) => {
+    setUsername(event.target.value)
+  }
+
+  const onEmailChange = (event) => {
+    setEmail(event.target.value)
+  }
+
+  const onPasswordChange = (event) => {
+    setPassword(event.target.value)
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault()
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ username, password, email })
+    }
+
+    fetch(API_URL('users'), options)
+      .then(res => res.json())
+      .then(data => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUsername(data.username))
+            dispatch(user.actions.setUserId(data.userId))
+            dispatch(user.actions.setAccessToken(data.accessToken))
+            dispatch(user.actions.setErrors(null))
+
+            localStorage.setItem('user', JSON.stringify({
+              userId: data.userId,
+              username: data.username,
+              accessToken: data.accessToken
+            }))
+          })
+        } else {
+          dispatch(user.actions.setErrors(data))
+        }
+      })
+  }
+
+  return (
+    <section className="login-container">
+      <form className="form-box-left" onSubmit={onFormSubmit}>
+        <h1 className="form-heading">Create an account</h1>
+        {errors && 
+          <>
+            {errors.error.code === 11000 
+            ? 
+            <p className="error-message">Email or username is not unique</p>
+            : 
+            <p className="error-message">{errors.message}</p>}
+          </>
+        }
+        <label className="input-wrapper">
+          <p className="input-label">Name</p>
+          <input
+            required
+            placeholder="User name"
+            className="input-box"
+            type="text"
+            value={username}
+            onChange={onNameChange}
+          />
+        </label>
+        <label className="input-wrapper">
+          <p className="input-label">Email</p>
+          <input
+            required
+            className="input-box"
+            placeholder="E-post"
+            type="email"
+            value={email}
+            onChange={onEmailChange}
+          />
+        </label>
+        <label className="input-wrapper">
+          <p className="input-label">password</p>
+          <input
+            required
+            className="input-box"
+            placeholder="Password"
+            type="password"
+            value={password}
+            onChange={onPasswordChange}
+          />
+        </label>
+        <div className="form-buttons-container">
+          <button type="submit" className="form-button">Sign up</button>
+        </div>
+      </form>
+      <div className="form-box-right">
+        <img src="./assets/moody.jpeg" alt="feeling bubbles" />
+      </div>
+    </section>
+  )
+}
```

Comment:
A Go Back button from sign up to first page would be good to have, in case someone clicked on signup by mistake.